src/main.cpp

```diff
@@ -977,8 +977,9 @@ class GraspProcessorModule : public RFModule
             //  precision of less than 1 cm is not accepted
             if (pose_candidates[0]->pose_cost_function(0) < 0.01)
             {
+                double min_pose_cost_function_0 = best_candidate->pose_cost_function(0);
                 best_candidate = pose_candidates[0];
-                for(size_t idx = 1; (idx < pose_candidates.size()) && (pose_candidates[idx]->pose_cost_function(0) < 0.01); idx++)
+                for (size_t idx = 1; (idx < pose_candidates.size()) && (pose_candidates[idx]->pose_cost_function(0) - min_pose_cost_function_0 < 0.002); idx++)
```

Comment:
The code -as it was before the PR- only considered candidates with position error smaller than 0.01m, and selected the candidate with the least orientation error cost. It is ok to check against the smallest position error but I don't think it will change much.

Reply:
Agree it's not resolutive, but we had the evidence of a wrong selection of the best candidate as per the pictures @towardthesea posted.